### DISCUSSION TITLE
Is ```harvest grafana import``` command still use?

### DISCUSSION THREAD

**Discussion by wooyoungAhn** [NetApp/harvest#2169] [Q&A] [answered]
Hello
I heard that grafana does not used api key. so can we still used comand ```harvest grafana import ```?
```
[root@harafana]# bin/harvest grafana import 
enter API token: 
```

**Comment by rahulguptajss** [comment]
@wooyoungAhn Do you mean to say that You can not generate API key? 

To generate an API key, follow these steps:

1. Navigate to the "API Keys" section in your application.

![image](https://github.com/NetApp/harvest/assets/25551691/42e57dca-0774-4e2d-97dd-6aabb7fb3262)

2. Generate a new API key with the "Editor" role assigned to it. Copy the generated key which now can be used in Harvest CLI.

![image](https://github.com/NetApp/harvest/assets/25551691/88a7924a-1204-4ff9-834a-76db048c5fb1)

**Comment by cgrinds** [comment]
hi @wooyoungAhn are you saying that Grafana has made a recent change to disallow API keys? Ah, I see I'm guessing you are talking about Grafana's note https://grafana.com/docs/grafana/latest/administration/service-accounts/

> Service accounts will eventually replace [API keys](https://grafana.com/docs/grafana/latest/administration/api-keys/) as the primary way to authenticate applications that interact with Grafana.

If so, API Keys still supported in Grafana and Harvest still uses them for [dashboard import/export](https://netapp.github.io/harvest/latest/dashboards/).

**Comment by wooyoungAhn** [comment]
Hello Thanks for your answer
In grafana 10 version this command does not work.
Maybe they have changed something?

```
Service account token created
glsa_XXXXXXXXXXXXXXXXXXXXXXXXXXXXXXXX_XXXXXXXX

[root@harafana]# bin/harvest grafana import 
enter API token:
no such host url=/api/folders/aaaaaaa
...

**Comment by rahulguptajss** [comment]
Thanks @wooyoungAhn for the information. Could you please provide the specific version of Grafana you are using? Additionally, could you clarify whether you are attempting to use the Editor role or the Admin role? I recommend trying the Admin role to see if you encounter the same error.

What is the output if you run this curl
`curl -H "Authorization: Bearer APIKEY" -X GET 'http://localhost:3000/api/folders/aaaaaaa'`

I tested versions 10.0.0 and 10.0.1 with the Admin privilege, and the Grafana import functionality worked without any issues. However, when using the Editor role, I encountered the following error, which indicates a permission problem that needs to be addressed in Harvest code:

`error connect: (403 - 403 Forbidden) You'll need additional permissions to perform this action. Permissions needed: folders:read
`

Please let me know the specific Grafana version you are using and whether the issue persists with the Admin role.

**Comment by rahulguptajss** [comment, marked as answer]
@wooyoungAhn The issue has been resolved in PR #2177. You can access the fix through the [nightly builds](https://github.com/NetApp/harvest/releases/tag/nightly).

**Reply by wooyoungAhn** [reply]
Thank you for your update
